src/gui/properties/trackerlist.cpp

```diff
@@ -394,22 +394,28 @@ void TrackerList::editSelectedTracker() {
     torrent->replaceTrackers(trackers);
     if (!torrent->isPaused()) {
       torrent->forceReannounce();
-      torrent->forceDHTAnnounce();
     }
 }
 
 #if LIBTORRENT_VERSION_NUM >= 10000
 void TrackerList::reannounceSelected() {
+    QList<QTreeWidgetItem *> selected_items = selectedItems();
+    if (selected_items.isEmpty()) return;
+
     BitTorrent::TorrentHandle *const torrent = properties->getCurrentTorrent();
     if (!torrent) return;
+    QList<BitTorrent::TrackerEntry> trackers = torrent->trackers();
 
-    QList<QTreeWidgetItem *> selected_items = getSelectedTrackerItems();
-    if (selected_items.isEmpty()) return;
+    foreach (QTreeWidgetItem* item, selected_items) {
+      // DHT case
+      if (item == dht_item) {
+        torrent->forceDHTAnnounce();
```

Comment:
I don't understand why you need to find the item and don't do `torrent->forceDHTAnnounce();` directly.